BookPlayer/Library/FileManagement/ImportOperation.swift

```
@@ -21,15 +22,67 @@ class ImportOperation: Operation {
+            let tempURL = file.destinationFolder.appendingPathComponent("tmp")
+
+            do {
+                try FileManager.default.createDirectory(at: tempURL, withIntermediateDirectories: true, attributes: nil)
+                try FileManager.default.unzipItem(at: file.originalUrl, to: tempURL)
+
+            } catch {
+                print("Extraction of ZIP archive failed with error:\(error)")
+            }
+
+            do {
+                let fileManager = FileManager.default
+
+                do {
+                    let resourceKeys: [URLResourceKey] = [.creationDateKey, .isDirectoryKey]
+                    let enumerator = fileManager.enumerator(at: tempURL,
+                                                            includingPropertiesForKeys: resourceKeys,
+                                                            options: [.skipsHiddenFiles], errorHandler: { (url, error) -> Bool in
+                                                                print("directoryEnumerator error at \(url): ", error)
+                                                                return true
+                    })!
+
+                    for case let fileURL as URL in enumerator {
+                        if fileURL.pathExtension == "mp3" || fileURL.pathExtension == "m4a" || fileURL.pathExtension == "m4b" {
+                            let documentsURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+                            let destinationURL = documentsURL.appendingPathComponent(fileURL.lastPathComponent)
+                            try FileManager.default.moveItem(at: fileURL, to: destinationURL)
+                        }
+                    }
+                } catch {
+                    print(error)
+                }
+
+                // Delete temp directory
+                do {
+                    try FileManager.default.removeItem(at: tempURL)
+                } catch {
```

same thing here, since no error handling is happening, use `try? ` to let it fail silently